internal/runnerinstall/nomad.go

```diff
@@ -73,13 +74,16 @@ func (i *NomadRunnerInstaller) Install(ctx context.Context, opts *InstallOpts) e
 		if i.Config.HostVolume != "" {
 			return fmt.Errorf("choose either CSI or host volume, not both")
 		}
+		if i.Config.CsiVolumeName == "" {
+			i.Config.CsiVolumeName = "waypoint-" + opts.Id + "-runner"
+		}
 
 		s = sg.Add("Creating persistent volume")
 		err = nomadutil.CreatePersistentVolume(
 			ctx,
 			client,
 			"waypoint-"+opts.Id+"-runner",
-			"waypoint-"+opts.Id+"-runner",
+			i.Config.CsiVolumeName,
```

Comment:
If the volume name is customizable, I think that the [way which we find the volume to clean up](https://github.com/hashicorp/waypoint/pull/3546/files#diff-d3925a67c54177f1d703959aa6e1bdce099cfdd2768b9155b621873b26e3781dR371) during `runner uninstall` might need to be updated as well. Currently for the uninstall, the volume name is tied to the Nomad job name.

Reply:
That's a great call out, and I also had worried about that and didn't know how to put the context into the commit but it looks like we only ever use the name when we setup the volume during creation. During uninstallation we look up volumes using the volume ID we provided the CSI provider at install time which has not been changed.

See https://github.com/hashicorp/waypoint/blob/main/internal/runnerinstall/nomad.go#L356 which adds the `prefix` url parameter when querying Nomad for a list of volumes. It doesn't look like its explicitly documented on Nomad's docs website but the `prefix` query parameter lists all volumes starting with the ID given. (also see it in nomad land: https://github.com/hashicorp/nomad/blob/main/nomad/csi_endpoint.go#L134-L135)